### DISCUSSION TITLE
How to pass accessToken to ApolloClient in NextjsApp Router ?

### DISCUSSION THREAD

**Discussion by lakshmajee** [logto-io/logto#5219] [General]
Logto provides helpers to access user info and to fetch accessToken in serverside.
But the application im working on right is using best of both worlds server side and client side (RSC &SSR) components.
Is there any way I can provide accessToken to ApolloClient request headers on client side?

**Comment by charIeszhao** [comment]
Hi, there's no out-of-the-box support for this scenario yet. Our SDK is either for client or server at the moment. Maybe you can implement a storage layer that can be accessed by both the client and server. @wangsijie Any other ideas?

**Reply by lakshmajee** [reply]
Thanks for response, 
But what should we do if the `client component` needs to make an API request with access token in `Authorization` header.
Side Note: Auth0 and Kinde works well in this case with Nextjs App router

**Reply by lakshmajee** [reply]
Is there any API available to fetch access token on serverside of Next.js ?

**Reply by wangsijie** [reply]
Are you using async component? Then you can get the access token in an async component, and then pass it to the client as "props".

**Reply by lakshmajee** [reply]
yes using async component but using app router, so no props right

**Reply by wangsijie** [reply]
There is an example file: https://github.com/logto-io/js/blob/master/packages/next-server-actions-sample/app/page.tsx

The file is an async component, and it can get context by calling `getLogtoContext`, and then, thoese values can pass down to client components. But it is unable to get access token for now, this is a bug and will be fixed soon.

**Reply by lakshmajee** [reply]
Will `getCookie` contains the user accessToken to be authorised by external service ?
Saw this code https://github.com/logto-io/js/blob/master/packages/next-server-actions-sample/libraries/logto.ts#L53C9-L53C18 and im unsure if this contains access token for user or not.
```ts
export const handleSignIn = async (searchParams: Record<string, string>) => {
  // Convert searchParams object into a query string.
  const search = new URLSearchParams(searchParams).toString();

  const newCookie = await logtoClient.handleSignInCallback(
    getCookie(),
    `${config.baseUrl}/callback?${search}`
  );

  setCookies(newCookie);
};


```

**Reply by wangsijie** [reply]
It contains access token, but it will expires in 1 hour by default. So I am afraid you have to wait for the new version, it will be released with-in several days.

**Reply by lakshmajee** [reply]
Sure. thank you @wangsijie 🙂

**Reply by wangsijie** [reply]
Sorry, my mistake, the SDK is OK, the problem is on the sample project, please see this PR: https://github.com/logto-io/js/pull/611
